src/modules/submodules/consensus.py

```diff
@@ -209,29 +205,29 @@ def get_blockstamp_for_report(self, last_finalized_blockstamp: BlockStamp) -> Op
             last_finalized_slot_number=last_finalized_blockstamp.slot_number,
         )
         logger.info({'msg': 'Calculate blockstamp for report.', 'value': bs})
+
+        # Make sure module is compatible with contracts on reference blockstamp.
+        self._check_contract_versions(bs)
         return bs
 
-    def _check_contract_versions(self, blockstamp: BlockStamp) -> bool:
+    def _check_contract_versions(self, blockstamp: ReferenceBlockStamp):
+        """
+        Check if Oracle can process report on reference blockstamp.
+        """
         contract_version = self.report_contract.get_contract_version(blockstamp.block_hash)
         consensus_version = self.report_contract.get_consensus_version(blockstamp.block_hash)
 
-        if contract_version > self.CONTRACT_VERSION or consensus_version > self.CONSENSUS_VERSION:
-            raise IncompatibleContractVersion(
-                f'Incompatible Oracle version. '
-                f'Expected contract version {contract_version} got {self.CONTRACT_VERSION}. '
-                f'Expected consensus version {consensus_version} got {self.CONSENSUS_VERSION}.'
-            )
-
-        compatibility = contract_version == self.CONTRACT_VERSION and consensus_version == self.CONSENSUS_VERSION
+        compatibility = (
+            contract_version in self.COMPATIBLE_CONTRACT_VERSIONS
+            and consensus_version in self.COMPATIBLE_CONSENSUS_VERSIONS
+        )
 
         if not compatibility:
-            logger.warning({
-                'msg': f'Oracle\'s version higher than contract supports. '
-                       f'Expected contract version {contract_version} got {self.CONTRACT_VERSION}. '
-                       f'Expected consensus version {consensus_version} got {self.CONSENSUS_VERSION}.'
-            })
-
-        return compatibility
+            raise IncompatibleOracleVersion(
+                f'Incompatible Oracle version. '
+                f'Expected contract version {contract_version} got {self.COMPATIBLE_CONTRACT_VERSIONS}. '
```

Comment:
I would swap the variables, in the way that we expect `COMPATIBLE_CONTRACT_VERSIONS` but got `contract_version` (because it's what we get from the network, right?).

Reply:
fixed